### ISSUE TITLE
Using PolySharap with Visual Studio Build Tools 2022

### ISSUE THREAD

**Issue by janis-veinbergs** [Sergio0694/PolySharp#70]
Is it possible to build PolySharp-enabled project using only Visual Studio Build Tools?. I'm on 17.5.4, yet it complains:

```
packages\PolySharp.1.13.1\build\PolySharp.targets(45,5): Error : The PolySharp source generators have been disabled on the current configuration, as they need Roslyn 4.3 in order to work. PolySharp requires the source generators to run in order to process shaders, so the library cannot be used without a more up to date IDE (eg. VS 2022 17.3 or greater) or .NET SDK version (.NET 6.0.400 SDK or greater).
```

I suspect it won't work without full IDE? Using Visual Studio Installer, I cannot find component "C# and Visual Basic Roslyn compilers" (Microsoft.VisualStudio.Component.Roslyn.LanguageServices). Can .NET6/7 somehow be used for language services when targeting .NET Framework? Or is it just for targeting older .NET Core versions? I have installed .NET6/7 on the host too, but the error is still there.

Perhaps if Build Tools are not supported, should be documented as it would help to decide to use PolySharp or no. it happens that my devops agent uses build tools and not the full IDE.

**Comment by janis-veinbergs** [comment]
Actually cannot confirm if this is working/not working with build tools - it happened that azure devops agent VSBuild task was still finding VS 2019, I did install VS2022 IDE version and remove VS2019, now it won't complain. But I suspect build tools would be still left out as I wasn't able to find "C# and Visual Basic Roslyn compilers" (Microsoft.VisualStudio.Component.Roslyn.LanguageServices) there.

Can anyone confirm PolySharp is compatible with Visual Studio Build Tools 2022 or not?

**Comment by saucecontrol** [comment]
The required MSBuild and Roslyn compiler versions are part of the .NET SDK.  As long as you have the current (7.0) SDK version, you can build from command line.

The latest SDK is only supported in the latest Visual Studio version, as listed on the [download page](https://dotnet.microsoft.com/en-us/download/dotnet/7.0), but to be clear, VS is not required at all to use PolySharp.

**Comment by janis-veinbergs** [comment]
Ok @saucecontrol, that would imply I'd have to migrate to sdk-style project. Thanks for answer.

**Comment by Sergio0694** [comment]
> "I'd have to migrate to sdk-style project"

You don't have to. PolySharp works just fine with legacy-style .csproj projects too.
The only requirements are that:
- You need to have the .NET 6+ SDK installed (as you'll need Roslyn 4.3+)
- You must be using `<PackageReference>` to reference PolySharp (`packages.config` isn't supported)

**Comment by janis-veinbergs** [comment]
> You must be using <PackageReference> to reference PolySharp (packages.config isn't supported)

That could be it. I did have .NET 6 and 7 installed. But turns out, project wasn't using PackageReferences. But I could still build this project with full IDE and  "C# and Visual Basic Roslyn compilers" (Microsoft.VisualStudio.Component.Roslyn.LanguageServices enabled (via Visual Studio Installer).

But I cannot test now the case as I moved on by installing full IDE on build server.
